src/main/scala/com/jakainas/functions/package.scala

```diff
@@ -169,25 +169,39 @@ package object functions {
   }
 
   /**
-   * Converts a timestamp into bigint (long) in milliseconds (default Spark returns only seconds)
-   *
-   * @param timestamp - Timestamp to extract milliseconds from
-   * @return Milliseconds since epoch for given timestamp
-   */
+    * Converts a timestamp into bigint (long) in milliseconds (default Spark returns only seconds)
+    *
+    * @param timestamp - Timestamp to extract milliseconds from
+    * @return Milliseconds since epoch for given timestamp
+    */
   def millis(timestamp: Column): Column = (timestamp.cast("double") * 1000).cast("bigint")
 
+  /**
+    * countDistinct equivalent for window operations.
+    * Distinct window operations are not currently supported by Spark so we can't just use countDistinct.
+    *
+    * TODO: see if performance can be improved using a UDAF or some other approach.
+    *
+    * @param countCol : Column values to count.
+    * @param partCols : One or more columns to partition the dataset by.
+    * @return : Count of unique values in countCol per partitionCols.
+    **/
+  def countUnique(countCol: Column, partCols: Column*): Column = {
```

Comment:
Nice addition.  I've used this pattern just this past week.

Same formatting comment as the other PR, good after.